addons/account_peppol/tests/test_peppol_participant.py

```diff
@@ -93,32 +105,38 @@ def test_create_participant_missing_data(self):
         with self.assertRaises(ValidationError), self.cr.savepoint():
             settings.button_create_peppol_proxy_user()
 
+    def test_create_participant_already_exists(self):
+        # creating a participant that already exists on Peppol network should not be possible
+        vals = self._get_participant_vals()
+        vals['account_peppol_eas'] = '0208'
+        settings = self.env['res.config.settings'].create(vals)
+        with self.assertRaises(UserError), self.cr.savepoint():
+            settings.button_create_peppol_proxy_user()
+
     def test_create_success_participant(self):
         # should be possible to apply with all data
         # the account_peppol_proxy_state should correctly change to pending
         # then the account_peppol_proxy_state should change success
         # after checking participant status
         company = self.env.company
         settings = self.env['res.config.settings'].create(self._get_participant_vals())
-
-        with self._patch_peppol_requests():
-            settings.button_create_peppol_proxy_user()
-            self.assertEqual(company.account_peppol_proxy_state, 'not_verified')
-            settings.button_send_peppol_verification_code()
-            self.assertEqual(company.account_peppol_proxy_state, 'sent_verification')
-            settings.account_peppol_verification_code = '123456'
-            settings.button_check_peppol_verification_code()
-            self.assertEqual(company.account_peppol_proxy_state, 'pending')
-            self.env['account_edi_proxy_client.user']._cron_peppol_get_participant_status()
-            self.assertEqual(company.account_peppol_proxy_state, 'active')
+        settings.button_create_peppol_proxy_user()
+        self.assertEqual(company.account_peppol_proxy_state, 'not_verified')
+        settings.button_send_peppol_verification_code()
+        self.assertEqual(company.account_peppol_proxy_state, 'sent_verification')
+        settings.account_peppol_verification_code = '123456'
+        settings.button_check_peppol_verification_code()
+        self.assertEqual(company.account_peppol_proxy_state, 'pending')
+        self.env['account_edi_proxy_client.user']._cron_peppol_get_participant_status()
+        self.assertEqual(company.account_peppol_proxy_state, 'active')
 
     def test_create_reject_participant(self):
         # the account_peppol_proxy_state should change to rejected
         # if we reject the participant
         company = self.env.company
         settings = self.env['res.config.settings'].create(self._get_participant_vals())
 
-        with self._patch_peppol_requests(reject=True):
+        with self._set_context({'reject': True}):
             settings.button_create_peppol_proxy_user()
```

Comment:
example of my first comment:
```suggestion
            settings.with_context(reject=True).button_create_peppol_proxy_user()
```